js/alerts.js

```diff
@@ -53,41 +100,177 @@ function loadSelectors() {
+                $('#platform').val(platform);
+            } catch (e) {
+                throw e;
+            }
+
+            //after changes are finalised, button is safe to click
+            $('#button').bind("click", filterOnClick).removeAttr('disabled');
+
+            //after the changes are made updated options are ready to be choosed
+            $('#test').removeAttr('disabled');
+            $('#platform').removeAttr('disabled');
+
+
+            if (callback && typeof(callback)==="function") 
+            {
+                callback();
+            }
+        });
+
+    }
+    else if(changedElementId == "test")
+    {
+        $('#test').removeAttr('disabled');
+        
+        $.getJSON(root_url+"/getvalues", $.param(get_params, true), function(data) {
+           
+            var tests = data['test'].sort(compare);
+            var revs = data['rev'];
+            var platforms = data['platform'].sort(compare);
+            
+            $("#rev").children().remove().end().append('<option value="">Select Revision</option>');
+            resetOptions(revs,"rev");
+            
+            $("#platform").children().remove().end().append('<option value="">Select Platform</option>');
+            resetOptions(platforms,"platform");
+
+            try {
+                $('#rev').val(rev);
+                $('#test').val(test);
+                $('#platform').val(platform);
+            } catch (e) {
+                throw e;
+            }
+            $('#button').bind("click", filterOnClick).removeAttr('disabled');
+            
+            $('#rev').removeAttr('disabled');
+            $('#platform').removeAttr('disabled');
+
+            if (callback && typeof(callback)==="function") 
+            {
+                callback();
+            }
+        });
+
+    }
+    else if(changedElementId == "platform")
+    {
+        $('#platform').removeAttr('disabled');
+        
+        $.getJSON(root_url+"/getvalues", $.param(get_params, true),function(data) {
+           
+            var tests = data['test'].sort(compare);
+            var revs = data['rev'];
+            var platforms = data['platform'].sort(compare);
+
+            $("#rev").children().remove().end().append('<option value="">Select Revision</option>');
+            resetOptions(revs,"rev");
+            
+            $("#test").children().remove().end().append('<option value="">Select Test</option>');
+            resetOptions(tests,"test");
+
+            try {
+                $('#rev').val(rev);
+                $('#test').val(test);
+                $('#platform').val(platform);
+            } catch (e) {
+                throw e;
+            }
+            $('#button').bind("click", filterOnClick).removeAttr('disabled');
+            
+            $('#rev').removeAttr('disabled');
+            $('#test').removeAttr('disabled');
+            
+            if (callback && typeof(callback)==="function") 
+            {
```

Comment:
nit: I prefer to see brackets on the same line, instead of:
if (cond)
{
  ...
}
else
{
 ...
}

it is cleaner to see:
if (cond) {
 ...
} else {
 ...
}